dlt/extract/incremental/transform.py

```diff
@@ -117,18 +120,49 @@ def find_cursor_value(self, row: TDataItem) -> Any:
 
         Will use compiled JSONPath if present, otherwise it reverts to column search if row is dict
         """
-        row_value: Any = None
         if self._compiled_cursor_path:
-            row_values = find_values(self._compiled_cursor_path, row)
-            if row_values:
-                row_value = row_values[0]
+            cursor_values = find_values(self._compiled_cursor_path, row)
+            if cursor_values == []:
+                if self.on_cursor_value_missing == "raise":
+                    raise IncrementalCursorPathMissing(self.resource_name, self.cursor_path, row)
+                elif self.on_cursor_value_missing == "exclude":
+                    return None
+            elif None in cursor_values:
+                if self.on_cursor_value_missing == "raise":
+                    raise IncrementalCursorPathHasValueNone(
+                        self.resource_name, self.cursor_path, row
+                    )
+                elif self.on_cursor_value_missing == "exclude":
+                    return None
+
+            # ignores the other found values, e.g. when the path is $data.items[*].created_at
+            row_value = cursor_values[0]
```

Comment:
IMO you should take the first element and then check for None. Also let's add this behavior to docs (if not yet added). Currently you check None in all possible locations

Reply:
Do you recommend taking the first element even before checking if the result is an empty list? In that case, I think we'd need to manage the control flow using try/except which is much slower than if/else so I'd avoid that.

The other alternative I see is flipping the order of checking for `None` and taking the first element. As follows:
```py
# ignores the other found values, e.g. when the path is $data.items[*].created_at
row_value = cursor_values[0]
if None in cursor_values:
    if self.on_cursor_value_missing == "raise":
        raise IncrementalCursorPathHasValueNone(
            self.resource_name, self.cursor_path, row
        )
    elif self.on_cursor_value_missing == "exclude":
        return None
```